src/Maestro/DependencyUpdater/DependencyUpdater.cs

```diff
@@ -217,32 +217,42 @@ private async Task CheckSubscriptionsAsync(UpdateFrequency targetUpdateFrequency
         {
             var enabledSubscriptionsWithTargetFrequency = (await Context.Subscriptions
                     .Where(s => s.Enabled)
-                    .Include(s => s.Channel)
-                    .ThenInclude(c => c.BuildChannels)
-                    .ThenInclude(bc => bc.Build)
-                    .ToListAsync())
-                .Where(s => s.PolicyObject.UpdateFrequency == targetUpdateFrequency);
+                    .ToListAsync(cancellationToken))
+                    .Where(s => s.PolicyObject.UpdateFrequency == targetUpdateFrequency);
 
             int subscriptionsUpdated = 0;
             foreach (var subscription in enabledSubscriptionsWithTargetFrequency)
             {
-                Build latestBuildInTargetChannel = subscription.Channel.BuildChannels.Select(bc => bc.Build)
+                Subscription subscriptionWithBuilds = await Context.Subscriptions
```

Comment:
What's a little crappy is that (before and here) we are likely looking up the same build channel over and over again. When arcade triggers in the morning to 100+ repos, it's executing the same query over and over again to find the latest build.

Reply:
I'm not super intimate with this EF code. It seems like just being able to express the query in EF is a trouble point. Is that right?

Reply:
I'm not sure it's that. Even if we made the original giant query into raw SQL so that the frequencies would be filtered database side, we were still getting a lot of duplicate data in the results when we try to map all the builds in the channel for each individual subscription, because many subscriptions target the same channel.

If we want to reduce the number of lookups of the same thing, I think I can:

1. Get the subscriptions we want and their target channels in one query and load that into memory.
2. Get the Builds from the latest x days for the channels that we got in another query and load that into memory.
3. iterate over all subscriptions and get the latest build for the repo we need in the appropriate channel's list.

Reply:
Yeah, essentially dedupe on the client side before making query 2. I like that approach